### PR TITLE
Migrate to JUnit Jupiter

### LINKED ISSUE #3135: Migrate from spock to normal junit (eventually jupiter)
Right now, E2E tests are using a different programming language, groovy, and a less used framework, spock. This complicates both the act of adding new tests as well dealing with dependencies.

At the time, it may have made sense to try something like Spock. However, if you look closely at the E2E tests, they are simple and neither need to use BDD style, nor need to use a different programming language.

Basic junit is easier for new maintainers and the code, and I believe this should be reverted to normal java. I'm happy to prove this in a PR if maintainers agree.

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This migrates to JUnit Jupiter with a bulk of help by OpenRewrite, but due to some use of systemstubs and wiremock, a significant amount of manual work also. It is surely the case this was easier with OpenRewrite despite the glitches.

Here are the main recipes that helped, though I couldn't run sequentially and in some cases I needed to roll back some changes.

```bash
$ mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-testing-frameworks:RELEASE -Drewrite.activeRecipes=org.openrewrite.java.testing.junit5.JUnit4to5Migration
$  mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.recipeArtifactCoordinates=org.openrewrite.recipe:rewrite-testing-frameworks:RELEASE -Drewrite.activeRecipes=org.openrewrite.java.testing.junit5.JUnit5BestPractices
# below I have to revert the changes to src/main as they are generated files
$ mvn -U org.openrewrite.maven:rewrite-maven-plugin:run -Drewrite.activeRecipes=org.openrewrite.java.RemoveUnusedImports
```

Here are notes on manual changes I did, most were functional, but some to remove distraction. Anecdotally, everything I did this morning and spent between 3 and 4 hours soup to nuts. This would have taken a lot longer without openrewrite.
* pom changes had to be redone
* most wiremock tests had to be edited manually due to use of static imports
* systemstubs had to be redone
* spring tests had to polished to not use redundant annotations
* temp directories had to be redone to simplify the result. this was in combination of moving from File to Path
* parameterized tests added some boilerplate looked better if was removed (think single test)
* some assertions became jupiter and had to be rolled back to assertj
* javadoc link to tests renamed from testX to x drift
* some test fixtures were copy pasted from other tests but not referenced. I removed these vs port them for no reason
* some tests were missing `@Test` in the first place so weren't run ever or in a long time. I added the annotation rather than delete those tests.

Fixes #3135

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
cc @timtebeek

File: examples/examples-release-18/src/test/java/io/kubernetes/client/examples/ExampleTest.java [hunk 1]
```diff
@@ -14,33 +14,36 @@
...
-  public void exactUrlOnly() throws ApiException {
+  void exactUrlOnly() throws ApiException {
     ApiClient client = new ApiClient();
     client.setBasePath("http://localhost:" + PORT);
     Configuration.setDefaultApiClient(client);
 
     V1Namespace ns1 = new V1Namespace().metadata(new V1ObjectMeta().name("name"));
 
-    stubFor(
+    apiServer.stubFor(
```
Comment (codefromthecrypt) [inline examples/examples-release-18/src/test/java/io/kubernetes/client/examples/ExampleTest.java:46]: this is an example which bit me.. basically when using the junit rule, you can mix/match using static method or the method on the rule. in the extension you must use the method and not use the static.

File: util/src/test/java/io/kubernetes/client/ExecTest.java [hunk 2]
```diff
@@ -46,25 +43,12 @@
...
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
 
 /** Tests for the Exec helper class */
-public class ExecTest {
-
-  public static class CountRequestAction extends PostServeAction {
```
Comment (codefromthecrypt) [inline util/src/test/java/io/kubernetes/client/ExecTest.java:56]: this was never referenced, so I deleted it

File: examples/examples-release-18/src/test/java/io/kubernetes/client/examples/ExampleTest.java (hunk 1 above)
Comment (codefromthecrypt) [reply examples/examples-release-18/src/test/java/io/kubernetes/client/examples/ExampleTest.java:46]: when you don't do this, things will fail about reaching port 8080 and you have to think through why. I think this maybe should become an openrewrite recipe for wiremock (removing the static method footgun and always even in junit 4 using the instance to do stubbing and verification cc @timtebeek)

**Comment by codefromthecrypt** [comment]
thanks for clicking run.

looks like problem on windows:
```
Error:  Errors: 
Error:    FilePersisterTest.persistence(Path) � IO Failed to delete temp directory C:\Users\RUNNER~1\AppData\Local\Temp\junit5731886460482893149. The following paths could not be deleted (see suppressed exceptions for details): <root>, testconfig1129927534881067784.tmp
Error:    KubeConfigTest.gcpAuthProvider(Path) � IO Failed to delete temp directory C:\Users\RUNNER~1\AppData\Local\Temp\junit1608042563520011234. The following paths could not be deleted (see suppressed exceptions for details): <root>, config15826931468867220118.tmp
Error:    KubeConfigTest.gcpAuthProviderStringDate(Path) � IO Failed to delete temp directory C:\Users\RUNNER~1\AppData\Local\Temp\junit15928002169510443350. The following paths could not be deleted (see suppressed exceptions for details): <root>, config14368825936767128865.tmp
```

and also some issue in LeaderElectorTest.

I'll try to bite these off in the next couple days as a change like this has massive drift work risk

**Comment by codefromthecrypt** [comment]
cool looks like green!

File: examples/examples-release-18/src/test/java/io/kubernetes/client/examples/ExampleTest.java (hunk 1 above)
Comment (timtebeek) [reply examples/examples-release-18/src/test/java/io/kubernetes/client/examples/ExampleTest.java:46]: Thanks for the suggestion! That's bit me before in the past as well; would be neat to have a recipe that changes any static method use to use through `rule` / `extension` ahead of any other WireMock migration steps. A good addition to this earlier issue.
- https://github.com/openrewrite/rewrite-testing-frameworks/issues/278

**Comment by brendandburns** [comment]
Thanks for the patience!

**Comment by timtebeek** [comment]
Great to see this merged! I've done a quick check through the platform and it seems there's two files that mere missed, perhaps as part of resolving a merge conflict. If desired a PR can be opened from this URL: https://app.moderne.io/results/KozquRLdd
```patch
diff --git a/util/src/test/java/io/kubernetes/client/CopyTest.java b/util/src/test/java/io/kubernetes/client/CopyTest.java
index 82fce66..3611992 100644
--- a/util/src/test/java/io/kubernetes/client/CopyTest.java
+++ b/util/src/test/java/io/kubernetes/client/CopyTest.java
@@ -183,7 +183,7 @@ io.moderne.worker.execution.RecipeMapper
   }
 
   @Test
-  void testCopyDirectoryFromPod(@TempDir Path tempDir) throws Exception {
+  void copyDirectoryFromPod(@TempDir Path tempDir) throws Exception {
     Copy copy = new Copy(client);
 
     apiServer.stubFor(
diff --git a/util/src/test/java/io/kubernetes/client/ExecCallbacksTest.java b/util/src/test/java/io/kubernetes/client/ExecCallbacksTest.java
index d150eb0..3e99fc9 100644
--- a/util/src/test/java/io/kubernetes/client/ExecCallbacksTest.java
+++ b/util/src/test/java/io/kubernetes/client/ExecCallbacksTest.java
@@ -13,7 +13,6 @@ io.moderne.worker.execution.RecipeMapper
 package io.kubernetes.client;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.jupiter.api.Assertions.fail;
 
 import io.kubernetes.client.custom.IOTrio;
 import io.kubernetes.client.openapi.ApiException;
@@ -31,6 +30,7 @@
 import java.util.concurrent.TimeoutException;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.function.BiConsumer;
+import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 class ExecCallbacksTest {
@@ -201,11 +201,9 @@
   }
 
   private void quietly(NoisyRunnable task) {
-    try {
+    Assertions.assertDoesNotThrow(() -> {
       task.run();
-    } catch (Exception e) {
-      fail(e);
-    }
+    });
   }
 
   private String read(InputStream is, String expectedText) {
@@ -219,13 +217,11 @@
   }
 
   private String read(InputStream is, int count) {
-    try {
+    Assertions.assertDoesNotThrow(() -> {
       byte[] buff = new byte[count];
       is.read(buff);
       return new String(buff);
-    } catch (IOException e) {
-      fail(e);
-    }
+    });
     return null;
   }
 ```

**Comment by codefromthecrypt** [comment]
hi @timtebeek. so looked intentionally doing something different, so I left it at `fail`. Also, if it weren't I would use assertj not Jupiter's assertThrows for consistency with everything else. I should have put a `TODO: why` on those, to save you the effort.

test renames are legit though!

**Pull request merged**